// @rspack/core/dist/builtin-plugin/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deprecated_resolveBuiltins = exports.RspackBuiltinPlugin = void 0;
var base_1 = require("./base");
Object.defineProperty(exports, "RspackBuiltinPlugin", { enumerable: true, get: function () { return base_1.RspackBuiltinPlugin; } });
__exportStar(require("./DefinePlugin"), exports);
__exportStar(require("./ProvidePlugin"), exports);
__exportStar(require("./BannerPlugin"), exports);
__exportStar(require("./ProgressPlugin"), exports);
__exportStar(require("./EntryPlugin"), exports);
__exportStar(require("./ExternalsPlugin"), exports);
__exportStar(require("./NodeTargetPlugin"), exports);
__exportStar(require("./ElectronTargetPlugin"), exports);
__exportStar(require("./HttpExternalsRspackPlugin"), exports);
__exportStar(require("./EnableChunkLoadingPlugin"), exports);
__exportStar(require("./EnableLibraryPlugin"), exports);
__exportStar(require("./EnableWasmLoadingPlugin"), exports);
__exportStar(require("./ChunkPrefetchPreloadPlugin"), exports);
__exportStar(require("./ArrayPushCallbackChunkFormatPlugin"), exports);
__exportStar(require("./CommonJsChunkFormatPlugin"), exports);
__exportStar(require("./ModuleChunkFormatPlugin"), exports);
__exportStar(require("./HotModuleReplacementPlugin"), exports);
__exportStar(require("./WebWorkerTemplatePlugin"), exports);
__exportStar(require("./WorkerPlugin"), exports);
__exportStar(require("./LimitChunkCountPlugin"), exports);
__exportStar(require("./MergeDuplicateChunksPlugin"), exports);
__exportStar(require("./SplitChunksPlugin"), exports);
__exportStar(require("./NamedModuleIdsPlugin"), exports);
__exportStar(require("./DeterministicModuleIdsPlugin"), exports);
__exportStar(require("./NamedChunkIdsPlugin"), exports);
__exportStar(require("./DeterministicChunkIdsPlugin"), exports);
__exportStar(require("./RealContentHashPlugin"), exports);
__exportStar(require("./RemoveEmptyChunksPlugin"), exports);
__exportStar(require("./EnsureChunkConditionsPlugin"), exports);
__exportStar(require("./WarnCaseSensitiveModulesPlugin"), exports);
__exportStar(require("./DataUriPlugin"), exports);
__exportStar(require("./FileUriPlugin"), exports);
__exportStar(require("./RuntimePlugin"), exports);
__exportStar(require("./JsonModulesPlugin"), exports);
__exportStar(require("./InferAsyncModulesPlugin"), exports);
__exportStar(require("./JavascriptModulesPlugin"), exports);
__exportStar(require("./AsyncWebAssemblyModulesPlugin"), exports);
__exportStar(require("./AssetModulesPlugin"), exports);
__exportStar(require("./SourceMapDevToolPlugin"), exports);
__exportStar(require("./EvalSourceMapDevToolPlugin"), exports);
__exportStar(require("./SideEffectsFlagPlugin"), exports);
__exportStar(require("./FlagDependencyExportsPlugin"), exports);
__exportStar(require("./FlagDependencyUsagePlugin"), exports);
__exportStar(require("./MangleExportsPlugin"), exports);
__exportStar(require("./HtmlRspackPlugin"), exports);
__exportStar(require("./CopyRspackPlugin"), exports);
__exportStar(require("./SwcJsMinimizerPlugin"), exports);
__exportStar(require("./SwcCssMinimizerPlugin"), exports);
function resolveTreeShaking(treeShaking, production) {
    return treeShaking !== undefined
        ? treeShaking.toString()
        : production
            ? "true"
            : "false";
}
function deprecated_resolveBuiltins(builtins, options, compiler) {
    var _a;
    const contextPath = options.context;
    const production = options.mode === "production" || !options.mode;
    const isRoot = !compiler.isChild();
    return {
        // TODO: discuss with webpack, this should move to css generator options
        css: options.experiments.css
            ? {
                modules: {
                    localsConvention: "asIs",
                    localIdentName: production
                        ? "[hash]"
                        : "[path][name][ext]__[local]",
                    exportsOnly: false,
                    ...(_a = builtins.css) === null || _a === void 0 ? void 0 : _a.modules
                }
            }
            : undefined,
        treeShaking: resolveTreeShaking(builtins.treeShaking, production)
    };
}
exports.deprecated_resolveBuiltins = deprecated_resolveBuiltins;
